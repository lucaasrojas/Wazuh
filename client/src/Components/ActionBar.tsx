import React from 'react'
import { Grid, Button } from '@mui/material'
import { Link } from 'react-router-dom'
const ActionBar : React.FunctionComponent  = () => {
    return (
        <Grid item xs={12} md={12} style={{display: "flex", justifyContent: "center"}}>
            <Link to="/users"><Button variant="text">Users</Button></Link>
            <Link to="/tasks"><Button variant="text">Tasks</Button></Link>
        </Grid>
    )
}

export default ActionBar
