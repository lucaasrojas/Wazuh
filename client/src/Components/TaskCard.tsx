
import React from 'react';
import { Grid } from '@mui/material'
import { makeStyles } from '@mui/styles';
import TaskInterface from 'Assets/Interfaces/TaskInterface'
const useStyles = makeStyles({
    wrapper: {
        flexFlow: "nowrap", overflow: "hidden" 
    },
    taskBullet: {
        width: 20, 
        height: "100%", 
        borderRadius: "4px 0px 0px 4px", 
        backgroundColor: (task : TaskInterface) => task.completed ? "green" : "gray"
    }
})

interface TaskCardProps {
    task: TaskInterface
}

const TaskCard : React.FunctionComponent<TaskCardProps>  = (props) => {
    const classes = useStyles(props.task)

    return props.task ? (
        <Grid key={props.task.id} container spacing={1} className={classes.wrapper}>
            <Grid item>
                <div className={classes.taskBullet}></div>
            </Grid>
            <Grid item>
                {props.task.title}
            </Grid>
        </Grid>
    ) : <></>
}

export default TaskCard;
