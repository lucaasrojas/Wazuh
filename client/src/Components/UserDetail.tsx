import React from 'react';
import { CardContent, Typography, Paper, Divider, Grid } from '@mui/material'
import { Link, useParams } from 'react-router-dom'
import { makeStyles } from '@mui/styles';
import { TaskCard } from '.'
import EmailIcon from '@mui/icons-material/Email';
import PersonalVideoIcon from '@mui/icons-material/PersonalVideo';
import ApartmentIcon from '@mui/icons-material/Apartment';
import PublicIcon from '@mui/icons-material/Public';
import * as api from '../Api/index'
import TaskInterface from 'Assets/Interfaces/TaskInterface'
import UserInterface from 'Assets/Interfaces/UserInterface'
interface DataCardInterface {
    icon: React.ReactElement,
    text: string | React.ReactElement
}

const DataCard : React.FunctionComponent<DataCardInterface> = ({ icon, text }) => {
    return (
        <Grid container spacing={2}>
            <Grid item xs={2} md={2}>
                {icon}
            </Grid>
            <Grid item xs={10} md={10}>
                <Typography sx={{ mb: 1.5 }} color="text.secondary">
                    {text}
                </Typography>
            </Grid>
        </Grid>

    )
}

interface QueryParams {
    id: string
}

const UserDetail = () => {
    const [userData, setUserData] = React.useState<UserInterface | undefined>()
    const [userTasks, setUserTasks] = React.useState<TaskInterface[]>([])
    const params : QueryParams = useParams()
    React.useEffect(() => {
        api.getUserById({ id: params.id })
            .then(res => setUserData(res))

        api.getTasksByUser({ id: params.id })
            .then(res => setUserTasks(res.data))
    }, [])


    return userData ? (

        <Grid container style={{ maxWidth: 480, margin: "0 auto", justifyContent: "center" }}>
            <Grid item xs={12} md={12}>
                <Typography variant="h6" style={{ fontSize: "100%", maxWidth: "100%" }} >
                    <strong>{userData.name}</strong>
                </Typography>
                <Divider style={{ margin: "10px 0px" }} />
            </Grid>

            <Grid item xs={12} md={12}>
                <Typography sx={{ mb: 1.5 }} color="text.secondary">
                    {/* {translate('contributor.fields.followers')}: {props.contributor.followers} */}
                    @{userData.username}
                </Typography>
            </Grid>

            <Grid item xs={12} md={6}>
                <DataCard icon={<EmailIcon />} text={userData.email} />
            </Grid>

            <Grid item xs={12} md={6}>
                <DataCard icon={<PersonalVideoIcon />} text={<a style={{ textDecoration: "none", color: "inherit" }} href={userData.website}>{userData.website}</a>} />
            </Grid>

            <Grid item xs={12} md={6}>
                <DataCard icon={<ApartmentIcon />} text={userData.company.name} />
            </Grid>

            <Grid item xs={12} md={6}>
                <DataCard icon={<PublicIcon />} text={userData.address.city} />

            </Grid>
            <Grid item container spacing={3} xs={12} md={12}>
                <Grid item xs={12} md={12}>
                    <Typography>
                        Tasks
                    </Typography>
                </Grid>
                {
                    userTasks.map((task : TaskInterface) => (
                        <Grid key={task.id} item xs={12} md={6}>
                            <TaskCard task={task} />
                        </Grid>
                    ))
                }
            </Grid>
        </Grid>
    ) : <></>
}

export default UserDetail
