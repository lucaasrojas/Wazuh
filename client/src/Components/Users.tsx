import React from 'react'
import * as api from '../Api/index'
import { Grid, Input } from '@mui/material'
import { UserCard } from './index'
import UserInterface from '../Assets/Interfaces/UserInterface'

const Users = () => {
    const [usersList, setUsersList] = React.useState<UserInterface[]>([])
    const [usersListFiltered, setUsersListFiltered] = React.useState<UserInterface[]>([])
    const [searchValue, setSearchValue] = React.useState<string | undefined>()
    React.useEffect(() => {
        api.getUsers()
            .then(({ data }) => {
                setUsersList(data)
                setUsersListFiltered(data)
            })
    }, [])

    React.useEffect(() => {
        if (searchValue) {
            const newArr = usersList.filter((el:UserInterface) =>el.name.toLowerCase().includes(searchValue) || el.email.toLowerCase().includes(searchValue) || el.username.toLowerCase().includes(searchValue))
            setUsersListFiltered(newArr)
        } else {
            setUsersListFiltered(usersList)
        }
    }, [searchValue])

    const handleSearch = (e : React.ChangeEvent<HTMLInputElement>) => {
        setSearchValue(e.target.value)
    }
    return (
        <Grid container spacing={3} style={{ maxWidth: 968, margin: "0 auto" }}>
            <Grid item xs={12} md={12} style={{ display: "flex", justifyContent: "center" }}>
                <Input value={searchValue} onChange={handleSearch} placeholder="Search" />
            </Grid>
            <Grid container spacing={3} item xs={12} md={12}>
                {
                    usersListFiltered.map((el : UserInterface) => (
                        <Grid key={el.id} item xs={12} sm={4} md={4}>
                            <UserCard user={el} />
                        </Grid>
                    ))
                }
            </Grid>
        </Grid>
    )
}

export default Users
