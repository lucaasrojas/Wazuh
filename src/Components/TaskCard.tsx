import React from 'react';
import { Grid } from '@mui/material'
import { makeStyles } from '@mui/styles';
import TaskInterface from 'Assets/Interfaces/TaskInterface'
const useStyles = makeStyles({
    wrapper: {
        flexFlow: "nowrap", overflow: "hidden" 
    },
    taskBullet: {
        width: 20, 
        height: "100%", 
        borderRadius: "4px 0px 0px 4px"
    }
})

interface TaskCardProps {
    task: TaskInterface
}

const TaskCard : React.FunctionComponent<TaskCardProps>  = (props) => {
    const classes = useStyles(props.task)

    return props.task ? (
        <Grid key={props.task.id} container spacing={1} className={classes.wrapper}>
            <Grid item>
                <div id={`task-${props.task.id}-bullet`} className={classes.taskBullet} style={{backgroundColor: props.task.completed ? "green" : "gray"}}></div>
            </Grid>
            <Grid item>
                {props.task.title}
            </Grid>
        </Grid>
    ) : <></>
}

export default TaskCard;
