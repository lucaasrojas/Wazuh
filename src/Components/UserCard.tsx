
import React from 'react';
import { CardContent, Typography, Paper, Divider } from '@mui/material'
import { Link } from 'react-router-dom'
import { makeStyles } from '@mui/styles';
import UserInterface from 'Assets/Interfaces/UserInterface'

const useStyles = makeStyles({
    wrapper: {
        textDecoration: "none",
    },
    paper: {
        "&:hover": {
            backgroundColor: "#EBEBEB",
            transform: "scale(1.03)",
            transition: 'transform .4s ease-in-out, background-color .4s ease-in-out'
        },
        textAlign: "center"
    },
    divider: {
        margin: "10px 0px"
    },
    nameLabel: {
        fontSize: "100%", maxWidth: "100%"
    }
})

interface UserCardProps {
    user: UserInterface
}

const UserCard : React.FunctionComponent<UserCardProps> = (props) => {
    const classes = useStyles()

    return props.user ? (
        <Link to={`/users/${props.user.id}`} className={classes.wrapper}>
            <Paper elevation={4} className={classes.paper}>
                <CardContent>
                    <Typography variant="h6" className={classes.nameLabel} >
                        <strong>{props.user.name}</strong>
                    </Typography>
                    <Divider className={classes.divider} />
                    <Typography sx={{ mb: 1.5 }} color="text.secondary">
                        @{props.user.username}
                    </Typography>
                    <Typography sx={{ mb: 1.5 }} color="text.secondary">
                        {props.user.email}
                    </Typography>
                </CardContent>
            </Paper>
        </Link>
    ) : <></>
}

export default UserCard;
