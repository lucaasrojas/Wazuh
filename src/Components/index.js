export { default as Users } from './Users/Users'
export { default as UserCard } from './UserCard/UserCard'
export { default as UserDetail } from './UserDetail/UserDetail'
export { default as Tasks } from './Tasks/Tasks'
export { default as TaskCard } from './TaskCard/TaskCard'
export { default as Layout } from './Layout/Layout'
export { default as ActionBar } from './ActionBar/ActionBar'
export { default as ResponsiveWrapper } from './ResponsiveWrapper/ResponsiveWrapper'