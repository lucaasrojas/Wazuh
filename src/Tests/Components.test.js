import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'
import { UserCard, TaskCard } from '../Components';
import 'regenerator-runtime/runtime'
import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import "@testing-library/jest-dom";
Enzyme.configure({
    adapter: new EnzymeAdapter()
})
jest.mock('axios');
jest.mock('../Api')

describe("UserCard component", () => {
    const mockedUser = {
        "id": 1,
        "name": "Leanne Graham",
        "username": "Bret",
        "email": "Sincere@april.biz",
        "address": {
            "street": "Kulas Light",
            "suite": "Apt. 556",
            "city": "Gwenborough",
            "zipcode": "92998-3874",
            "geo": {
                "lat": "-37.3159",
                "lng": "81.1496"
            }
        },
        "phone": "1-770-736-8031 x56442",
        "website": "hildegard.org",
        "company": {
            "name": "Romaguera-Crona",
            "catchPhrase": "Multi-layered client-server neural-net",
            "bs": "harness real-time e-markets"
        }
    }
    test('Should render', async () => {
        const wrapper = render(
            <BrowserRouter>
                <UserCard user={mockedUser} />
            </BrowserRouter>
        )
        expect(wrapper).toMatchSnapshot()
    });

    test('Should render with the correct data', async () => {
        const wrapper = render(
            <BrowserRouter>
                <UserCard user={mockedUser} />
            </BrowserRouter>
        )
        const cardTitle = wrapper.container.querySelector("strong").innerHTML;
        expect(cardTitle).toBe(mockedUser.name);
    });

    test('Without data should not render', async () => {
        const wrapper = render(
            <BrowserRouter>
                <UserCard />
            </BrowserRouter>
        )
        const cardTitle = wrapper.container.querySelector("strong");
        expect(cardTitle).toBeNull();
    });
})

describe("Task Card", () => {
    const mockedTask = {
        "user_id": 1,
        "id": 1,
        "title": "delectus aut autem",
        "completed": true
    }
    const incompleteTask = {
        "user_id": 1,
        "id": 2,
        "title": "delectus aut autem",
        "completed": false
    }
    test('The bullet color should be green for completed tasks', async () => {
        const wrapper = render(
            <TaskCard task={mockedTask} />
        )
        const bullet = wrapper.container.querySelector(`#task-${mockedTask.id}-bullet`)
        expect(getComputedStyle(bullet).backgroundColor).toBe('green')
    });

    test('The bullet color should be gray for incompleted tasks', async () => {
        const wrapper = render(
            <TaskCard task={incompleteTask} />
        )
        const bullet = wrapper.container.querySelector(`#task-${incompleteTask.id}-bullet`)
        expect(getComputedStyle(bullet).backgroundColor).toBe('gray')
    });

})