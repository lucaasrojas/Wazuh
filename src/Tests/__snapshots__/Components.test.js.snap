// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="makeStyles-wrapper-1"
        href="/users/1"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 makeStyles-paper-2 css-11mde6h-MuiPaper-root"
        >
          <div
            class="makeStyles-card-7"
          >
            <div
              class="makeStyles-titleWrapper-4"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 makeStyles-nameLabel-5 css-2ulfj5-MuiTypography-root"
              >
                <strong>
                  Leanne Graham
                </strong>
              </h6>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth makeStyles-divider-3 css-9mgopn-MuiDivider-root"
              />
            </div>
            <div
              class="makeStyles-cardContent-8"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 makeStyles-label-6 css-9eg9l8-MuiTypography-root"
              >
                @
                Bret
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 makeStyles-label-6 css-9eg9l8-MuiTypography-root"
              >
                Sincere@april.biz
              </p>
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="makeStyles-wrapper-1"
      href="/users/1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 makeStyles-paper-2 css-11mde6h-MuiPaper-root"
      >
        <div
          class="makeStyles-card-7"
        >
          <div
            class="makeStyles-titleWrapper-4"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 makeStyles-nameLabel-5 css-2ulfj5-MuiTypography-root"
            >
              <strong>
                Leanne Graham
              </strong>
            </h6>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth makeStyles-divider-3 css-9mgopn-MuiDivider-root"
            />
          </div>
          <div
            class="makeStyles-cardContent-8"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-label-6 css-9eg9l8-MuiTypography-root"
            >
              @
              Bret
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-label-6 css-9eg9l8-MuiTypography-root"
            >
              Sincere@april.biz
            </p>
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
